syntax = "proto3";

package face_recognition;

service FaceRecognition {
  // Register a face with student ID
  rpc RegisterFace(RegisterRequest) returns (RegisterResponse) {}
  
  // Identify a face and return student ID
  rpc IdentifyFace(IdentifyRequest) returns (IdentifyResponse) {}

  // Delete face embedding by student ID
  rpc DeleteFace(DeleteFaceRequest) returns (DeleteFaceResponse) {}
}

message RegisterRequest {
  string student_id = 1;
  string image_base64 = 2;
}

message RegisterResponse {
  bool success = 1;
  string message = 2;
}

message IdentifyRequest {
  string image_base64 = 1;
}

message IdentifyResponse {
  string student_id = 1;
  float confidence = 2;
  bool success = 3;
  string message = 4;
}

message DeleteFaceRequest {
  string student_id = 1;
}

message DeleteFaceResponse {
  bool success = 1;
  string message = 2;
}
